src/smolagents/default_tools.py

```diff
@@ -138,48 +138,62 @@ class GoogleSearchTool(Tool):
     }
     output_type = "string"
 
-    def __init__(self):
+    def __init__(self, provider: Optional[str] = None):
```

Comment:
Why not a `str` which defaults to "serpapi"?
```suggestion
    def __init__(self, provider: str = "serpapi"):
```